src/pages/settings/PasswordPage.js

```diff
@@ -51,57 +50,109 @@ class PasswordPage extends Component {
+            },
         };
 
-        this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.submit = this.submit.bind(this);
+        this.getErrorText = this.getErrorText.bind(this);
+        this.validate = this.validate.bind(this);
+        this.clearErrorAndSetValue = this.clearErrorAndSetValue.bind(this);
         this.currentPasswordInputRef = null;
+
+        this.errorKeysMap = {
+            currentPassword: 'passwordPage.errors.currentPassword',
+            confirmNewPassword: 'passwordPage.errors.confirmNewPassword',
+            newPasswordSameAsOld: 'passwordPage.errors.newPasswordSameAsOld',
+            confirmPasswordMatch: 'setPasswordPage.passwordsDontMatch',
+            newPassword: 'passwordPage.errors.newPassword',
+        };
     }
 
     componentWillUnmount() {
         clearAccountMessages();
     }
 
-    onBlurNewPassword() {
-        const stateToUpdate = {};
-        if (!this.state.newPassword || !this.isValidPassword()) {
-            stateToUpdate.isPasswordRequirementsVisible = true;
-        } else {
-            stateToUpdate.isPasswordRequirementsVisible = false;
+    /**
+     * @param {String} field
+     * @returns {String}
+     */
+    getErrorText(field) {
+        if (this.state.errors[field]) {
+            return this.props.translate(this.errorKeysMap[field]);
         }
+        return '';
+    }
 
-        if (this.state.newPassword && this.state.confirmNewPassword && !this.doPasswordsMatch()) {
-            stateToUpdate.shouldShowPasswordConfirmError = true;
-        } else {
-            stateToUpdate.shouldShowPasswordConfirmError = false;
+
+    /**
+     * @param {String} field
+     * @param {String} value
+     * @param {String[]} additionalErrorsToClear
+     */
+    clearErrorAndSetValue(field, value, additionalErrorsToClear) {
+        const errorsToReset = {
+            [field]: false,
+        };
+        if (additionalErrorsToClear) {
+            _.each(additionalErrorsToClear, (errorFlag) => {
+                errorsToReset[errorFlag] = false;
+            });
         }
 
-        this.setState(stateToUpdate);
+        this.setState(prevState => ({
+            [field]: value,
+            errors: {...prevState.errors, ...errorsToReset},
+        }));
     }
 
-    onBlurConfirmPassword() {
-        if ((this.state.newPassword && !this.state.confirmNewPassword) || !this.doPasswordsMatch()) {
-            this.setState({shouldShowPasswordConfirmError: true});
-        } else {
-            this.setState({shouldShowPasswordConfirmError: false});
+    /**
+     * @returns {Boolean}
+     */
+    validate() {
+        const errors = {};
+
+        if (!this.state.currentPassword) {
+            errors.currentPassword = true;
         }
-    }
 
-    isValidPassword() {
-        return this.state.newPassword.match(CONST.PASSWORD_COMPLEXITY_REGEX_STRING);
-    }
+        if (!this.state.newPassword || !isValidPassword(this.state.newPassword)) {
```

Comment:
NAB, `!this.state.newPassword` check should be performed inside `isValidPassword()`?